### PR TITLE
ocamlPackages.secp256k1-internal: init at 0.2.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Motivation for this change

More packages for ligo

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/secp256k1-internal/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    sha256 = "1g9fyi78nmmm19l2cggwj14m4n80rz7gmnh1gq376zids71s6qxv";
+  };
+
+  useDune2 = true;
+
+  propagatedBuildInputs = [
+    gmp
+    cstruct
+    bigstring
+    dune-configurator
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/secp256k1-internal/default.nix:28]: I guess that `dune-configurator` need not be propagated.

**Review by vbgl — COMMENTED** [review]
Please set `minimalOCamlVersion` to at least `"4.08"`.

**Comment by ulrikstrid** [comment]
For future reference, when do I need to set the `minimalOCamlVersion`?

File: pkgs/development/ocaml-modules/secp256k1-internal/default.nix (hunk 1 above)
Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/secp256k1-internal/default.nix:33]: Thank you, moved it to `buildInputs` since it won't build without it

**Comment by vbgl** [comment]
The `ocaml-ng` package set contains `ocamlPackages_N_NN` package sets corresponding to different versions of the OCaml compiler. Now that you’ve set `minimalOCamlVersion`, if you try to *evaluate* the attribute `ocaml-ng.ocamlPackages_4_05.secp256k1-internal`, you’ll get an error (this library is not available for that version of OCaml). Without the `minimalOCamlVersion` being set, evaluation will succeed but if you try to build the derivation, you’ll get a build-time error (because this library requires some functionalities of the language or of the standard library that are were not available with OCaml 4.05).

Build-time errors usually reveal bugs in `nixpkgs`: it is wrong to add a derivation that fails to build. Setting `minimalOCamlVersion` prevents this.

**Comment by ulrikstrid** [comment]
Okay, thank you @vbgl!

**Pull request merged**